packages/block-library/src/navigation-link/edit.js

```diff
@@ -196,8 +229,11 @@ function NavigationLinkEdit( {
 											normalizedTitle !== normalizedURL &&
 											label !== newTitle ) {
 											return escape( newTitle );
+										} else if ( label ) {
+											return label;
 										}
-										return label;
+										// If there's no label, add the URL.
+										return escape( normalizedURL );
 									} )(),
```

Comment:
Very weird code style, but this was here before anyway.

Reply:
Yeah, just trying to sneak in what I needed in the way that made the most sense :) I'm happy to redo it if you think it'd be cleaner a different way.